exercises/array_while_loops.md

```diff
@@ -0,0 +1,51 @@
+## Loops and arrays: more fun with fruit
+
+In this exercise, we'll use an alternative way of looping through all the items of an array. Let's see a case where it might be useful.
+
+#### Part 1
+
+We'll start with an empty unordered html list with the id "list" in the static html:
+
+```html
+<ul id="list"></ul>
+```
+
+1. As before, create an array called `fruits` of your favourite fruit inside an `at the start` block.
+
+We're going to use this array of `fruits` to build a HTML list.
+
+Instead of the `for each item in array` block, we will use the `repeat while` block and the `... is empty` block together
+
+2. Add a `find the element with id` block to find the html list.
+
+3. Add a `repeat while` block from the Loops menu inside the previous block and change it to be `repeat until`. (We are going to keep `get and remove`ing items from the array _until_ it's empty.)
+
+4. Set the condition value to `... is empty`.
+
+5. Inside the loop, add the blocks to create a new `<li>` and set the contents to a `get and remove the first item from the array` block, using the "fruits" array.
+
+6. <span class="test-checkbox"></span>Check that all your fruit are now displaying.
+
+#### Part 2
+
+Next, let's have each `<li>` contain two things. The name and the emoji of the fruit.
+
+We can only set the text contents of an element once. However, we can add multiple elements inside the same `<li>`. In html, the `<span>` element is typically used for such a purpose, because it doesn't add any extra meaning or formatting to the text (unlike others which say "this is a list item" or "this is a link")
```

Comment:
Not quite sure what you mean here: "We can only set the text vontes of an element once". Why only once ?

Reply:
It means you can't set the text content to 2 different things independantly, like "hello" and "world" (and also, there is no string concatenation block)